src/proxy/src/test/java/ee/ria/xroad/common/util/healthcheck/HealthChecksTest.java

```diff
@@ -0,0 +1,296 @@
+ * be in a certain state before the tests are run (and they don't set the conf up themselves)
+ */
+public class HealthChecksTest {
+
+    @Test
+    public void cacheResultOnceShouldCacheOnce() {
+
+        // prepare
+        final HealthCheckResult expectedFirstResult = failure("message for failure");
+        final HealthCheckResult expectedSecondResult = failure("some other message");
+
+        final HealthCheckProvider mockProvider = mock(HealthCheckProvider.class);
+        when(mockProvider.get()).thenReturn(expectedSecondResult);
+
+        final HealthCheckProvider testedProvider = HealthChecks.cacheResultOnce(mockProvider, expectedFirstResult);
+
+        // execute & verify
+        assertEquals("first result does not match", expectedFirstResult, testedProvider.get());
+        assertEquals("second result does not match", expectedSecondResult, testedProvider.get());
+        // next results should come from the provider, too
+        assertEquals("third result does not match", expectedSecondResult, testedProvider.get());
+    }
+
+    @Test
+    public void cacheResultForShouldCacheSuccessForSetTime() throws InterruptedException {
+
+        // prepare
+        final HealthCheckResult expectedFirstResult = OK;
+        final HealthCheckResult expectedSecondResult = expectedFirstResult;
+        final HealthCheckResult expectedThirdResult = failure("third result");
+
+        final int okCache = 1;
+        final int errorCache = 2;
+
+        // Mockito 1.10.19 does not seem to work with interface default methods:
+        // stackoverflow.com/questions/27663252/can-you-make-mockito-1-10-17-work-with-default-methods-in-interfaces
+        // wrapping this mockProvider with the tested cacheResultFor method returns null
+        // final HealthCheckProvider mockProvider = mock(HealthCheckProvider.class);
+        // when(mockProvider.get()).thenReturn(expectedFirstResult, expectedThirdResult);
+
+        final HealthCheckProvider mockProvider = mockToReturn(expectedFirstResult, expectedThirdResult);
+
+        final HealthCheckProvider testedProvider = mockProvider.map(
+                cacheResultFor(okCache, errorCache, TimeUnit.SECONDS));
+
+        // execute & verify
+        assertEquals("first result does not match", expectedFirstResult, testedProvider.get());
+        assertEquals("second result does not come from the cache", expectedSecondResult, testedProvider.get());
+        Thread.sleep(okCache * 1000L + 500L);
+        assertEquals("third result does come from the cache", expectedThirdResult, testedProvider.get());
+    }
+
+    @Test
+    public void cacheResultForShouldCacheErrorForSetTime() throws InterruptedException {
+
+        // prepare
+        final HealthCheckResult expectedFirstResult = failure("first result");
+        final HealthCheckResult expectedSecondResult = expectedFirstResult;
+        final HealthCheckResult expectedThirdResult = OK;
+
+        final int okCache = 1;
```

Comment:
Are these purely constants? Would uppercase naming OK_CACHE make it more obvious?